### ISSUE TITLE
[BUG] Text Input Max Length Fails WCAG 3.3.3 Error Suggestion

### ISSUE THREAD

**Issue by LukeTaylor34** [UKHO/admiralty-design-system#131]
**Describe the bug**
The max length text input is not accessible for non-sighted users as they are not notified when they type more than the max length and screen readers will continue to read out the characters they are typing without knowing that the field has stopped accepting characters. 

This also effects users ability to copy or autofill text as they could paste text into the input and not realise not all the text copied over if the copied text is more than the max length limit.

**Expected behaviour**
My thoughts are text inputs should not stop users adding characters to the input but should error when a user enters more than the max length limit. Suggested error message '{Field name} must be {Max length} characters or less.'. 

The error should appear once a user types over the max length limit or when they paste text more than the max limit rather than on change in focus.

**Additional context**
3.3.3: Error Suggestion - If an input error is automatically detected and suggestions for correction are known, then the suggestions are provided to the user, unless it would jeopardize the security or purpose of the content.

**Comment by KatiePUX** [comment]
remove the max length prop and put that onus on the developer
